frida_mode/src/persistent/persistent_arm64.c

```diff
@@ -107,16 +107,16 @@ static void instrument_persitent_save_regs(GumArm64Writer *  cw,
   /* Q */
   gum_arm64_writer_put_stp_reg_reg_reg_offset(
```

Comment:
Looks like ARM64 has 32 of these Q registers... https://github.com/capstone-engine/capstone/blob/fee83fcc1ad096c22d4f2066ccb58ad1a76a9886/include/capstone/arm64.h#L453

Gum meanwhile provides 16 `v` registers (each 16 bytes in size)...
https://github.com/frida/frida-gum/blob/eb4faac39fa2db6d221d329ee50ca2c8059543f1/gum/gumdefs.h#L259

Thus I think the index into the `v` array should increment by `1` for each pair of registers rather than `4`, and we are missing pushing the remainder of the Q registers. Given the increased number of registers, this series of similar function calls should probably be replaced by a loop.

Reply:
I don't know whats a beautiful way to tackle this problem. I just came up with:

`  /* Q */
  // https://d-capstone.dpldocs.info/capstone.arm64.arm64_reg.html
  int reg_Q = 104;
  for (int i = 0; i < 16; i++) {
    gum_arm64_writer_put_stp_reg_reg_reg_offset(
      cw, (arm64_reg)reg_Q, (arm64_reg)(reg_Q + 1), ARM64_REG_X0,
      offsetof(GumCpuContext, v[i]), GUM_INDEX_SIGNED_OFFSET);
      reg_Q = reg_Q + 2;
  }`

Reply:
Yeah. I don't think a beautiful way exists, but that's along the lines I was thinking...

```
for (int i = 0; i < 16; i++) {

    gum_arm64_writer_put_stp_reg_reg_reg_offset(
         cw, ARM64_REG_Q0 + (i*2), ARM64_REG_Q0 + (i*2) + 1, ARM64_REG_X0,
         offsetof(GumCpuContext, v[i]), GUM_INDEX_SIGNED_OFFSET);

}
```